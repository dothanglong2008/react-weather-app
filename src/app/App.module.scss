* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

html {
  font-size: calc(10px + 1vmin);
}

ul {
  list-style: none;
}

img,
picture,
figure {
  width: 100%;
  object-fit: none;
}

button,
a {
  cursor: pointer;
}

.app {
  background-color: #ebebeb;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
}
