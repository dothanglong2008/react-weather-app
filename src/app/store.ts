import { configureStore, ThunkAction, Action } from "@reduxjs/toolkit";
import weatherReducer from "../features/weather/weatherSlice";
import searchFormReducer from "../components/weather/search/searchFormSlice";

export const store = configureStore({
  reducer: {
    weather: weatherReducer,
    searchForm: searchFormReducer,
  },
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
export type AppStore = typeof store;

export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
