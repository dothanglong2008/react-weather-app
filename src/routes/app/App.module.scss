$background-color-primary: #ffffff;
$text-color-primary: #000000;
$font-size-primary: 16px;

* {
  margin: 0;
  padding: 0;
  width: 100%;
}

body {
  background-color: $background-color-primary;
  color: $text-color-primary;
  font-size: $font-size-primary;
}

ul {
  list-style: none;
}

img,
picture,
video {
  max-width: 100%;
}
