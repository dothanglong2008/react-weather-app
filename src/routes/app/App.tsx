import Item from "../../components/weather/single-item/item/Item";
import styles from "./App.module.scss";

function App() {
  return (
    <div className={styles.app}>
      <Item />
    </div>
  );
}

export default App;
