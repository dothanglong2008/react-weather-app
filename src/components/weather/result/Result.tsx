import HourlyWeather from "../hourly-weather/HourlyWeather";
import DailyWeather from "../daily-weather/DailyWeather";
import useAutoRefreshWeather from "../../../hooks/useAutoRefreshWeather";
import { useAppSelector } from "../../../app/hooks";
import { selectLocation } from "../../../features/weather/weatherSlice";
import TextBox from "../../ui/elements/text-box/TextBox";
import SectionContainer from "../../ui/containers/section-container/SectionContainer";

function Result() {
  useAutoRefreshWeather();
  const location = useAppSelector(selectLocation);

  return location ? (
    <>
      <SectionContainer>
        <HourlyWeather />
      </SectionContainer>
      <SectionContainer>
        <DailyWeather />
      </SectionContainer>
    </>
  ) : (
    <SectionContainer>
      <TextBox content={<>Search for a city or get current location</>} />
    </SectionContainer>
  );
}

export default Result;
