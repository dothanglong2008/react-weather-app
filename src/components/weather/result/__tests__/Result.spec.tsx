import renderer from "react-test-renderer";
import Result from "../Result";
import { Provider } from "react-redux";
import { store } from "../../../../app/store";

describe("Test Result component", () => {
  it("renders Result correctly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Result />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
