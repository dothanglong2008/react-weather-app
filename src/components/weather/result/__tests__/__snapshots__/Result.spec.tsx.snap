// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Result component renders Result correctly 1`] = `
<section
  className="section"
>
  <p>
    Search for a city or get current location
  </p>
</section>
`;
